lib/spack/spack/directives.py

```diff
@@ -259,6 +262,29 @@ def variant(pkg, name, default=False, description=""):
     pkg.variants[name] = Variant(default, description)
 
 
+@directive('resources')
+def resource(pkg, **kwargs):
+    """
+    Define an external resource to be fetched and staged when building the package. Based on the keywords present in the
+    dictionary the appropriate FetchStrategy will be used for the resource.
+
+    List of recognized keywords:
+
+    * 'when' : represents the condition upon which the resource is needed (optional)
+    * 'destination' : path where to extract / checkout the resource (optional)
+
+    """
+    when = kwargs.get('when', pkg.name)
+    # FIXME : currently I assume destination to be a relative path (rooted at pkg.stage.source_path)
+    destination = kwargs.get('destination', "")
+    when_spec = parse_anonymous_spec(when, pkg.name)
+    resources = pkg.resources.setdefault(when_spec, [])
+    # FIXME : change URLFetchStrategy with a factory that selects based on kwargs
+    fetcher = URLFetchStrategy(**kwargs)
```

Comment:
Would it be fine to have a factory in `fetch_strategy.py` that accepts and parses `kwargs` and returns the appropriate strategy? If so should it be a static method of `FetchStrategy`? I mean something like:
```python
class FetchStrategy(object):
    @staticmethod
    def create(**kwargs):
        # Selects strategy based on required arguments
        return strategy
```
The idea would be to handle cases in which resources are checked-out from repositories, e.g. if a support for `@trunk` will be added to clang.